src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -0,0 +1,326 @@
+import static seedu.address.logic.parser.RecipeBookSyntax.PREFIX_TAG;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_RECIPES;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.RecipeBookParserUtil;
+import seedu.address.model.Model;
+import seedu.address.model.recipe.CompletionTime;
+import seedu.address.model.recipe.Ingredient;
+import seedu.address.model.recipe.Name;
+import seedu.address.model.recipe.Recipe;
+import seedu.address.model.recipe.ServingSize;
+import seedu.address.model.recipe.Step;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Edits a recipe to the recipe book.
+ */
+public class EditCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the recipe"
+            + " identified by the index number used in the displayed recipe list. "
+            + "Existing values will be overwritten by the input values.\n\n"
+            + "Compulsory parameters (only name OR index is necessary):\n"
+            + "NAME (name of recipe to be edited)\n"
+            + PREFIX_INDEX + "INDEX (index of recipe to be edited)\n\n"
+            + "Optional Parameters:\n"
+            + PREFIX_NAME + "NAME\n"
+            + PREFIX_COMPLETION_TIME + "COMPLETION TIME\n"
+            + PREFIX_SERVING_SIZE + "SERVING SIZE\n"
+            + "[" + PREFIX_INGREDIENT + "INGREDIENT]...\n"
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "[" + PREFIX_STEP + "STEP]...\n\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_INDEX + "1 "
+            + PREFIX_NAME + "Aglio Olio "
+            + PREFIX_COMPLETION_TIME + "10 "
+            + PREFIX_SERVING_SIZE + "1 "
+            + PREFIX_INGREDIENT + "Spaghetti "
+            + PREFIX_INGREDIENT + "Garlic "
+            + PREFIX_INGREDIENT + "Sauce "
+            + PREFIX_INGREDIENT + "Bacon "
+            + PREFIX_STEP + "Cook the pasta "
+            + PREFIX_STEP + "Saute the garlic "
+            + PREFIX_STEP + "Toss pasta in the sauce "
+            + PREFIX_STEP + "Add bacon in pasta "
+            + PREFIX_STEP + "Taste and season "
+            + PREFIX_TAG + "Italian "
+            + PREFIX_TAG + "Western";
+
+    public static final String MESSAGE_EDIT_RECIPE_SUCCESS = "Edited Recipe: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_RECIPE = "This recipe already exists in the recipe book.";
+
+    private final Optional<Name> targetName;
+    private final Optional<Index> targetIndex;
```

Comment:
Good use of optional, we may have to update how View and Delete handles this as it currently assumes a null value.